### PR TITLE
Removes redundant else in registry/session.go

### LINKED ISSUE #12523: Remove redunant else clause
`else` should be removed in https://github.com/docker/docker/blob/master/registry/session.go#L227 since we are returning from the previous `if`. Split the `else if` clause in a separate `if` below.

### PR DESCRIPTION
**Pull request by RickWieman**
Very small contribution; fixes #12523.

### CODE REVIEW

File: registry/session.go [hunk 1]
```diff
@@ -224,8 +224,6 @@ func (r *Session) GetRemoteTags(registries []string, repository string, token []
 
 		if res.StatusCode == 404 {
 			return nil, fmt.Errorf("Repository not found")
-		} else if res.StatusCode != 200 {
```
Comment (thaJeztah) [inline registry/session.go]: Oh! I think the original description wasn't clear on this; the actual "continue" should still be executed, but in a separate "if", (no "else if"), i.e.:

```go
if res.StatusCode != 200 {
    continue
}
```

**Comment by thaJeztah** [comment]
NOT LGTM (sorry for the confusion)

You can update this PR by amending your last commit and a forced push, this should update this PR

File: registry/session.go (hunk 1 above)
Comment (duglin) [inline registry/session.go]: it'll be interesting to see if any of the tests fail  :-)

**Comment by RickWieman** [comment]
Whoops, my bad. Fixed it :wink:

File: registry/session.go (hunk 1 above)
Comment (thaJeztah) [inline registry/session.go]: With the "wrong" change? Hm, too late for that, the PR is updated, LOL

Comment (RickWieman) [inline registry/session.go]: Sorry for following up so fast :grin:

**Comment by thaJeztah** [comment]
No worries, the description wasn't clear :)

**Comment by thaJeztah** [comment]
LGTM (but I'm not the maintainer), thanks!

**Comment by duglin** [comment]
LGTM

File: registry/session.go (hunk 1 above)
Comment (runcom) [inline registry/session.go]: I've seen janky passing before the new updated commit :)

**Comment by LK4D4** [comment]
Personally I think it should be switch-case.
But LGTM for now.

**Pull request merged**